### PR TITLE
SUS-3421 | ChangesList::insertUserRelatedLinks - use either rc_user or rc_user_text for handling logged-in / anon entries in recentchanges table

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-3241

Compare http://sandbox-s2.nordycka.wikia.com/wiki/Specjalna:Ostatnie_zmiany vs http://nordycka.wikia.com/wiki/Specjalna:Ostatnie_zmiany

sus-3210 - recent changes

### CODE REVIEW

File: includes/ChangesList.php [hunk 1]
```diff
@@ -361,9 +361,13 @@ public function insertUserRelatedLinks( &$s, &$rc ) {
 		if( $this->isDeleted( $rc, Revision::DELETED_USER ) ) {
 			$s .= ' <span class="history-deleted">' . wfMsgHtml( 'rev-deleted-user' ) . '</span>';
```
Comment (WikiaTech) [inline includes/ChangesList.php]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

[project-id]: wikia:app

**Comment by WikiaTech** [comment]
SonarQube analysis reported 97 issues
* ![MAJOR][MAJOR] 73 major
* ![MINOR][MINOR] 8 minor
* ![INFO][INFO] 16 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WallHelper.class.php#L6](https://github.com/Wikia/app/blob/008baed597bd76b17108721bfa37085c20979338/extensions/wikia/Wall/WallHelper.class.php#L6): Class "WallHelper" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L74](https://github.com/Wikia/app/blob/008baed597bd76b17108721bfa37085c20979338/extensions/wikia/Wall/WallHelper.class.php#L74): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L232](https://github.com/Wikia/app/blob/008baed597bd76b17108721bfa37085c20979338/extensions/wikia/Wall/WallHelper.class.php#L232): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L260](https://github.com/Wikia/app/blob/008baed597bd76b17108721bfa37085c20979338/extensions/wikia/Wall/WallHelper.class.php#L260): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L384](https://github.com/Wikia/app/blob/008baed597bd76b17108721bfa37085c20979338/extensions/wikia/Wall/WallHelper.class.php#L384): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L487](https://github.com/Wikia/app/blob/008baed597bd76b17108721bfa37085c20979338/extensions/wikia/Wall/WallHelper.class.php#L487): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L561](https://github.com/Wikia/app/blob/008baed597bd76b17108721bfa37085c20979338/extensions/wikia/Wall/WallHelper.class.php#L561): Reduce the number of returns of this function 5, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L582](https://github.com/Wikia/app/blob/008baed597bd76b17108721bfa37085c20979338/extensions/wikia/Wall/WallHelper.class.php#L582): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L606](https://github.com/Wikia/app/blob/008baed597bd76b17108721bfa37085c20979338/extensions/wikia/Wall/WallHelper.class.php#L606): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [ChangesList.php#](https://github.com/Wikia/app/blob/008baed597bd76b17108721bfa37085c20979338/includes/ChangesList.php#): File "ChangesList.php" has 1335 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: includes/ChangesList.php [hunk 2]
```diff
@@ -808,8 +819,14 @@ public function lineLinksCache($rc, $unpatrolled, $counter) {
 			$out['userlink'] = ' <span class="history-deleted">' . wfMsgHtml( 'rev-deleted-user' ) . '</span>';
```
Comment (WikiaTech) [inline includes/ChangesList.php]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

[project-id]: wikia:app

**Comment by WikiaTech** [comment]
SonarQube analysis reported 97 issues
* ![MAJOR][MAJOR] 73 major
* ![MINOR][MINOR] 8 minor
* ![INFO][INFO] 16 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WallHelper.class.php#L6](https://github.com/Wikia/app/blob/e6558afc89f73a5ab227ee824a49f94bd9e4419d/extensions/wikia/Wall/WallHelper.class.php#L6): Class "WallHelper" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L74](https://github.com/Wikia/app/blob/e6558afc89f73a5ab227ee824a49f94bd9e4419d/extensions/wikia/Wall/WallHelper.class.php#L74): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L232](https://github.com/Wikia/app/blob/e6558afc89f73a5ab227ee824a49f94bd9e4419d/extensions/wikia/Wall/WallHelper.class.php#L232): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L260](https://github.com/Wikia/app/blob/e6558afc89f73a5ab227ee824a49f94bd9e4419d/extensions/wikia/Wall/WallHelper.class.php#L260): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L384](https://github.com/Wikia/app/blob/e6558afc89f73a5ab227ee824a49f94bd9e4419d/extensions/wikia/Wall/WallHelper.class.php#L384): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L487](https://github.com/Wikia/app/blob/e6558afc89f73a5ab227ee824a49f94bd9e4419d/extensions/wikia/Wall/WallHelper.class.php#L487): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L561](https://github.com/Wikia/app/blob/e6558afc89f73a5ab227ee824a49f94bd9e4419d/extensions/wikia/Wall/WallHelper.class.php#L561): Reduce the number of returns of this function 5, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L582](https://github.com/Wikia/app/blob/e6558afc89f73a5ab227ee824a49f94bd9e4419d/extensions/wikia/Wall/WallHelper.class.php#L582): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L606](https://github.com/Wikia/app/blob/e6558afc89f73a5ab227ee824a49f94bd9e4419d/extensions/wikia/Wall/WallHelper.class.php#L606): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [ChangesList.php#](https://github.com/Wikia/app/blob/e6558afc89f73a5ab227ee824a49f94bd9e4419d/includes/ChangesList.php#): File "ChangesList.php" has 1348 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: includes/ChangesList.php [hunk 3]
```diff
@@ -361,9 +361,14 @@ public function insertUserRelatedLinks( &$s, &$rc ) {
 		if( $this->isDeleted( $rc, Revision::DELETED_USER ) ) {
 			$s .= ' <span class="history-deleted">' . wfMsgHtml( 'rev-deleted-user' ) . '</span>';
```
Comment (WikiaTech) [inline includes/ChangesList.php:362]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

[project-id]: wikia:app

**Comment by WikiaTech** [comment]
SonarQube analysis reported 97 issues
* ![MAJOR][MAJOR] 73 major
* ![MINOR][MINOR] 8 minor
* ![INFO][INFO] 16 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WallHelper.class.php#L6](https://github.com/Wikia/app/blob/5b5d59a201c3a1919564e71acb1101efc5ba5e87/extensions/wikia/Wall/WallHelper.class.php#L6): Class "WallHelper" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L74](https://github.com/Wikia/app/blob/5b5d59a201c3a1919564e71acb1101efc5ba5e87/extensions/wikia/Wall/WallHelper.class.php#L74): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L232](https://github.com/Wikia/app/blob/5b5d59a201c3a1919564e71acb1101efc5ba5e87/extensions/wikia/Wall/WallHelper.class.php#L232): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L260](https://github.com/Wikia/app/blob/5b5d59a201c3a1919564e71acb1101efc5ba5e87/extensions/wikia/Wall/WallHelper.class.php#L260): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L384](https://github.com/Wikia/app/blob/5b5d59a201c3a1919564e71acb1101efc5ba5e87/extensions/wikia/Wall/WallHelper.class.php#L384): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L487](https://github.com/Wikia/app/blob/5b5d59a201c3a1919564e71acb1101efc5ba5e87/extensions/wikia/Wall/WallHelper.class.php#L487): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L561](https://github.com/Wikia/app/blob/5b5d59a201c3a1919564e71acb1101efc5ba5e87/extensions/wikia/Wall/WallHelper.class.php#L561): Reduce the number of returns of this function 5, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L582](https://github.com/Wikia/app/blob/5b5d59a201c3a1919564e71acb1101efc5ba5e87/extensions/wikia/Wall/WallHelper.class.php#L582): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WallHelper.class.php#L606](https://github.com/Wikia/app/blob/5b5d59a201c3a1919564e71acb1101efc5ba5e87/extensions/wikia/Wall/WallHelper.class.php#L606): Replace or remove this deprecated Wikia::log logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [ChangesList.php#](https://github.com/Wikia/app/blob/5b5d59a201c3a1919564e71acb1101efc5ba5e87/includes/ChangesList.php#): File "ChangesList.php" has 1348 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: includes/ChangesList.php (hunk 2 above)
Comment (WikiaTech) [inline includes/ChangesList.php:819]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

[project-id]: wikia:app

**Review by mszabo-wikia — APPROVED** [review]

**Pull request merged**